src/ui/src/core_components/other/CoreJsonViewer.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+	<div>
+		<p v-if="error" class="error">JSON Viewer error: {{ error }}</p>
+		<WdsJsonViewer
+			v-else-if="data"
+			ref="rootInstance"
+			:data="data"
+			:paths-open="pathsOpen"
+			@toggle="onToggle"
+		/>
+		<WdsJsonViewerChildrenCounter v-else :data="{}" />
+	</div>
+</template>
+
+<script lang="ts">
+import { cssClasses, separatorColor } from "../../renderer/sharedStyleFields";
+import { JsonViewerTogglePayload } from "../../wds/WdsJsonViewer.utils";
+import { FieldCategory, FieldType } from "../../writerTypes";
+
+const description = "A standalone element to view a JSON data.";
+
+const initialData = {
+	name: "JSON Viewer",
+	description: "A JSON tree viewer were you can expand key",
+	sample: {
+		description: "This sample is opened by default",
+		bool: true,
+		null: null,
+		list: [1, "two", { key: 3 }],
+	},
+	sampleClosed: {
+		description: "This sample is not opened by default",
+	},
+	createdAt: new Date(),
+};
+
+const toggleHandlerStub = `
+def handle_json_tree_toggle(state, payload):
+
+	# string array representing the keys of the path
+	path = payload.get("path")
+
+
+	# boolean if the path was open (True) or closed (False)
+	open = payload.get("open")
+
+	# ...`;
+
+export default {
+	writer: {
+		name: "JSON Viewer",
+		description,
+		category: "Other",
+
+		events: {
+			"wf-toggle": {
+				desc: "Capture JSON tree element open.",
+				stub: toggleHandlerStub.trim(),
+			},
+		},
+
+		fields: {
+			data: {
+				name: "Data",
+				init: JSON.stringify(initialData),
+				type: FieldType.Text,
```

Comment:
This should be a `FieldType.Object` which parses automatically to JSON. Also make sure to check that it works with state from backend e.g. `@{my_state}`

Reply:
Great, I wasn't aware of this